### ISSUE TITLE
[Suggestion/Discussion] Some Simple Nickname Tweaks

### ISSUE THREAD

**Issue by username11421** [Monika-After-Story/MonikaModDev#7562]
- A user on the MAS reddit pointed out that they couldn't nickname Monika "Cassey" due to the` "(?<![blmprs])ass(?!i)",` key. This would also block names like "Cassidy", "Lass", "Lassey", etc. Maybe one or two letters, but not more, in front of the problem characters, and several following them, could be allowed so as to allow these nicknames?
- Can "adulterer"/"adultery" be added to the `mas_bad_nickname_list`? As "cheater" is already on there. I also propose banning the nickname "harlot", but only in a way that won't prevent nicknames like "Charlotte".

**Comment by ThePotatoGuy** [comment]
sometimes I feel like we're better off not filtering nicknames at all. Is it more immersion breaking when she accepts a bad name or when she rejects a good name?

**Comment by username11421** [comment]
Personally, I think her accepting a negative name is more immersion breaking than declining a good name. If she declines a good name, it makes more sense that she just dislikes it versus a bad name being accepted.

**Comment by ThePotatoGuy** [comment]
"declines a good name" is not an accurate description of her response for "Cassidy" matching "ass". 

```renpy
m 4efd "[player]! That's not nice at all!"
m 2efc "Why would you say such things?"
m 2rfw "If you didn't want to do this, you should've just said so!"
m 2dftdc "..."
m 2ektsc "...You didn't have to be so mean."
m 2dftdc "That really hurt, [player]."
```

Not to mention that the nickname option gets locked after too many names that trigger the bad list.

**Comment by username11421** [comment]
> "declines a good name" is not an accurate description of her response for "Cassidy" matching "ass".
> 
> ```renpy
> m 4efd "[player]! That's not nice at all!"
> m 2efc "Why would you say such things?"
> m 2rfw "If you didn't want to do this, you should've just said so!"
> m 2dftdc "..."
> m 2ektsc "...You didn't have to be so mean."
> m 2dftdc "That really hurt, [player]."
> ```
> 
> Not to mention that the nickname option gets locked after too many names that trigger the bad list.

Good point. Unintentional triggers of that response are something to avoid.

**Comment by Booplicate** [comment]
The more feedback we're getting, the better the system's becoming. Allowing any names would be terrible, I'd rather add a huge whitelist of allowed nicknames, since calling her by names of other people is weird anyway. As for the player, we can ignore occasional bad names.

**Comment by username11421** [comment]
> since calling her by names of other people is weird anyway.

That's a good point and I agree with it - but looking at the MAS reddit, it seems quite a few people use other peoples' names for Monika. There is also some issue with when a name is also a noun/adjective e.g. "Hope", "Bonita", "Temperance", "Felicity" etc

**Comment by lunulae** [comment]
I would be open to figuring out if we could find a solution for people who have ended up locking the nickname system on accident. That has always seemed a rather glaring issue to me. But I agree that I prefer consistently adding to the whitelist to not having a bad names list at all.

**Comment by Booplicate** [comment]
We do plan to add another "second chance" for those who lost the option. At a cost, of course.
